src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.Reflection.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System;
+using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static partial class ActivatorUtilities
+    {
+#if NET8_0_OR_GREATER // Use the faster ConstructorInvoker which also has alloc-free APIs when <= 4 parameters.
+        private static object ReflectionFactory_ServiceOnly_Fixed(
```

Comment:
```suggestion
        private static object ReflectionFactoryServiceOnlyFixed(
```

https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md - (13)

> We use PascalCasing for all method names, including local functions.

Reply:
What does `ServiceOnly` mean?

Reply:
GetService() is called for each parameter.